### PR TITLE
Print exception trace for functions submitted to thread_pool

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
Currently we don't have any error log for functions submitted to thread_pool by `thread_pool.submit`, this PR is to print those error traces.

* Log before change:
```
TIMEOUT: //src/python/grpcio_tests/tests/unit:_abort_test.native (Summary)
      /usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/testlogs/src/python/grpcio_tests/tests/unit/_abort_test.native/test.log
INFO: From Testing //src/python/grpcio_tests/tests/unit:_abort_test.native:
==================== Test output for //src/python/grpcio_tests/tests/unit:_abort_test.native:
test_abort (__main__.AbortTest) ... -- Test timed out at 2023-01-11 01:03:36 UTC --
================================================================================
```

* Log After change:
```
TIMEOUT: //src/python/grpcio_tests/tests/unit:_abort_test.native (Summary)
      /usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/testlogs/src/python/grpcio_tests/tests/unit/_abort_test.native/test.log
INFO: From Testing //src/python/grpcio_tests/tests/unit:_abort_test.native:
==================== Test output for //src/python/grpcio_tests/tests/unit:_abort_test.native:
test_abort (__main__.AbortTest) ... Traceback (most recent call last):
  File "/usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/sandbox/linux-sandbox/1624/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.native.runfiles/com_github_grpc_grpc/src/python/grpcio/grpc/_server.py", line 495, in _
call_behavior
    response_or_iterator = behavior(argument, context)
  File "/usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/sandbox/linux-sandbox/1624/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.native.runfiles/com_github_grpc_grpc/src/python/grpcio_tests/tests/unit/_abort_test.py"
, line 54, in abort_unary_unary
    servicer_context.abort(
  File "/usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/sandbox/linux-sandbox/1624/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.native.runfiles/com_github_grpc_grpc/src/python/grpcio/grpc/_server.py", line 364, in a
bort
    raise Exception()
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/sandbox/linux-sandbox/1624/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.native.runfiles/com_github_grpc_grpc/src/python/grpcio/grpc/_server.py", line 622, in _
unary_response_in_pool
    response, proceed = _call_behavior(rpc_event, state, behavior,
  File "/usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/sandbox/linux-sandbox/1624/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.native.runfiles/com_github_grpc_grpc/src/python/grpcio/grpc/_server.py", line 500, in _
call_behavior
    _abort(state, rpc_event.call, cygrpc.StatusCode.unknown,
  File "/usr/local/google/home/xuanwn/.cache/bazel/_bazel_xuanwn/36b8ff3f6248f9f6b372e5787162cc70/sandbox/linux-sandbox/1624/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.native.runfiles/com_github_grpc_grpc/src/python/grpcio/grpc/_server.py", line 188, in _
abort
    raise Exception('test exception')
Exception: test exception
-- Test timed out at 2023-01-11 01:02:53 UTC --
================================================================================
```

### CODE REVIEW

File: src/python/grpcio/grpc/_server.py [hunk 1]
```diff
@@ -662,6 +666,8 @@ def send_response(response: Any) -> None:
                 if proceed:
                     _send_message_callback_to_blocking_iterator_adapter(
                         rpc_event, state, send_response, response_iterator)
+    except Exception:  # pylint: disable=broad-except
+        traceback.print_exception(*sys.exc_info())
```
Comment (gnossen) [inline src/python/grpcio/grpc/_server.py:670]: Maybe [`traceback.print_exc()`](https://docs.python.org/3/library/traceback.html#traceback.print_exc)

**Comment by gnossen** [comment]
Since there likely isn't a test for this, can you please manually verify the the behavior of this change and post the results here?

**Comment by XuanWang-Amos** [comment]
> Since there likely isn't a test for this, can you please manually verify the the behavior of this change and post the results here?

Sure, will add results to description.

**Comment by XuanWang-Amos** [comment]
@gnossen PTAL

**Review by gnossen — APPROVED** [review]

**Pull request merged**